hazelcast-sql-core/src/main/java/com/hazelcast/sql/impl/calcite/validate/operators/ReplaceUnknownOperandTypeInference.java

```diff
@@ -51,6 +51,10 @@ public ReplaceUnknownOperandTypeInference(SqlTypeName[] typeNames, SqlTypeName d
 
     @Override
     public void inferOperandTypes(SqlCallBinding callBinding, RelDataType returnType, RelDataType[] operandTypes) {
+        if (callBinding.getCall().isCountStar()) {
```

Comment:
I do not like that `COUNT(*)` leaks into a general-purpose `ReplaceUnknownOperandTypeInference`. Let's use another specialized implementation for `ReplaceUnknownOperandTypeInference` instead.